holidays/countries/micronesia.py

```diff
@@ -0,0 +1,209 @@
+        * [Truk](https://web.archive.org/web/20250214144551/http://fsmlaw.org/chuuk/code/title01/T01_Ch06.htm)
+
+        According to Section 602:
+            All holidays set forth in section 601 shall, if they occur on a Saturday,
+            be observed on the preceding Friday, and shall, if they occur on a Sunday,
+            be observed on the following Monday.
+
+        According to Kosrae State Legislature State Laws
+            * 4-32: Government Holidays were amended.
+            * 4-135: September 8 was declared a public holiday.
+            * 7-27: Good Friday was declared a public holiday.
+            * 7-76: Gospel Day was declared a public holiday.
+            * 9-50: Thanksgiving Day was declared a public holiday.
+            * 13-79: Kosrae Disability Day was declared a public holiday.
+
+        According to Pohnpei State Code 2012 Section 7-101:
+            * Pohnpei Consititution Day is a public holiday.
+            * Liberation Day is a public holiday.
+            * Pohnpei Cultural Day is a public holiday.
+            * Good Friday is a public holiday.
+
+    """
+
+    country = "FM"
+    default_language = "en_FM"
+    # %s (observed).
+    observed_label = tr("%s (observed)")
+    # Federated States of Micronesia gained independence on November 3, 1986
+    start_year = 1987
+    subdivisions = (
+        "TRK",  # Chuuk.
+        "KSA",  # Kosrae.
+        "PNI",  # Pohnpei.
+        "YAP",  # Yap.
+    )
+    subdivisions_aliases = {
+        "Chuuk": "TRK",
+        "Kosrae": "KSA",
+        "Pohnpei": "PNI",
+        "Yap": "YAP",
+    }
+    _deprecated_subdivisions = (
+        "Truk",
+        "Ponape",
+        "Kusaie",
+    )
+    supported_languages = ("en_FM", "en_US")
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        kwargs.setdefault("observed_rule", SAT_TO_PREV_FRI + SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("New Year's Day")))
+
+        # Micronesian Culture and Tradition Day.
+        name = tr("Micronesian Culture and Tradition Day")
+        if self._year >= 2010:
+            self._add_observed(self._add_holiday_mar_31(name))
+
+        # Federated States of Micronesia Day.
+        self._add_observed(self._add_holiday_may_10(tr("Federated States of Micronesia Day")))
+
+        if self._year >= 1991:
+            # United Nations Day.
+            self._add_observed(self._add_united_nations_day(tr("United Nations Day")))
+
+        # Independence Day.
+        self._add_observed(self._add_holiday_nov_3(tr("Independence Day")))
+
+        if self._year >= 2004:
+            # FSM Veterans of Foreign Wars Day.
+            self._add_observed(self._add_remembrance_day(tr("FSM Veterans of Foreign Wars Day")))
+
+        if self._year > 2020:
```

Comment:
The same goes for the ones in Kosrae
```suggestion
        if self._year >= 2021:
```